haystack_algolia/algolia_backend.py

```diff
@@ -296,9 +297,23 @@ def build_query(self):
             return '*'
 
         # no field filtering. It's an Algolia's feature, not a bug.
-        queries = [value.children[0][1] for value in self.query_filter.children]
+        return self._build_sub_query(self.query_filter)
 
-        return ' '.join(queries)
+    def _build_sub_query(self, search_node):
```

Comment:
Fantastic! Could you document a bit more of what this method is used for and what the different branches in the method body are doing?

Reply:
Let me know if it's understandable enough